x/pairing/keeper/unresponsive_provider.go

```diff
@@ -83,32 +83,41 @@ func (k Keeper) UnstakeUnresponsiveProviders(ctx sdk.Context, epochsNumToCheckCU
 		}
 	}
 	// Go over the staked provider entries (on all chains)
-	for _, providerStakeStorage := range providerStakeStorageList {
-		providerStakeEntriesForChain := providerStakeStorage.GetStakeEntries()
-		for _, providerStakeEntry := range providerStakeEntriesForChain {
-			if minHistoryBlock < providerStakeEntry.StakeAppliedBlock {
-				// this staked provider has too short history (either since staking
-				// or since it was last unfrozen) - do not consider for jailing
-				continue
-			}
-			// update the CU count for this provider in providerCuCounterForUnreponsivenessMap
-			providerPaymentStorageKeyList, complaintCU, servicedCU, err := k.countCuForUnresponsiveness(ctx, minPaymentBlock, epochsNumToCheckCUForUnresponsiveProvider, epochsNumToCheckCUForComplainers, providerStakeEntry)
+	pecsDetailed := k.GetAllProviderEpochCuStore(ctx)
+	for _, pec := range pecsDetailed {
+		if pec.ProviderEpochCu.ComplainersCu == 0 {
+			continue
+		}
+		providerAddr, err := sdk.AccAddressFromBech32(pec.Provider)
+		if err != nil {
+			continue
+		}
+		stakeEntry, found, _ := k.epochStorageKeeper.GetStakeEntryByAddressCurrent(ctx, pec.ChainId, providerAddr)
```

Comment:
we know that this fetching can be heavy, i think we have a list of all the providers above

Reply:
done. see https://github.com/lavanet/lava/commit/04683cabd599f9983112eb2a951f4ca2e04e70a5